backend/windmill-api/src/slack_approvals.rs

```diff
@@ -0,0 +1,1133 @@
+pub struct SlackFormData {
+    payload: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+struct Container {
+    message_ts: String,
+    channel_id: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Payload {
+    actions: Option<Vec<Action>>,
+    view: Option<View>,
+    trigger_id: Option<String>,
+    #[serde(rename = "type")]
+    r#type: String,
+    container: Option<Container>,
+}
+
+#[derive(Deserialize, Debug)]
+struct View {
+    state: Option<State>,
+    private_metadata: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Action {
+    value: Option<String>,
+    action_id: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct State {
+    values: HashMap<String, HashMap<String, ValueInput>>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type", rename_all = "snake_case")]
+enum ValueInput {
+    PlainTextInput { value: Option<Value> },
+    Datepicker { selected_date: Option<String> },
+    Timepicker { selected_time: Option<String> },
+    StaticSelect { selected_option: Option<SelectedOption> },
+    RadioButtons { selected_option: Option<SelectedOption> },
+    Checkboxes { selected_options: Option<Vec<SelectedOption>> },
+}
+
+#[derive(Deserialize, Debug)]
+struct SelectedOption {
+    value: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ResumeSchema {
+    schema: Schema,
+}
+
+#[derive(Debug, Deserialize)]
+struct ResumeFormRow {
+    resume_form: Option<serde_json::Value>,
+    hide_cancel: Option<bool>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Schema {
+    order: Vec<String>,
+    required: Vec<String>,
+    properties: HashMap<String, ResumeFormField>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ResumeFormField {
+    #[serde(rename = "type")]
+    r#type: String,
+    format: Option<String>,
+    default: Option<serde_json::Value>,
+    description: Option<String>,
+    title: Option<String>,
+    #[serde(rename = "enum")]
+    r#enum: Option<Vec<String>>,
+    #[serde(rename = "enumLabels")]
+    enum_labels: Option<HashMap<String, String>>,
+    nullable: Option<bool>,
+}
+
+#[derive(Deserialize)]
+pub struct QueryMessage {
+    message: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct QueryResourcePath {
+    slack_resource_path: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct QueryChannelId {
+    channel_id: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct QueryFlowStepId {
+    flow_step_id: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct ModalActionValue {
+    w_id: String,
+    job_id: String,
+    path: String,
+    approver: Option<String>,
+    message: Option<String>,
+    flow_step_id: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct PrivateMetadata {
+    resume_url: String,
+    resource_path: String,
+    container: Container,
+}
+
+pub async fn slack_app_callback_handler(
+    authed: Option<ApiAuthed>,
+    Extension(db): Extension<DB>,
+    Form(form_data): Form<SlackFormData>,
```

Comment:
I'm not sure but I think this should work
```suggestion
    Json(payload): Json<Payload>,
```
Or `From(payload): Form<Payload>` ?

Reply:
This query actually comes from the slack server and not from the windmill client and the data actually comes as form data (application/x-www-form-urlencoded) and not json (application/json)